src/main/java/org/openrewrite/java/testing/mockito/PowerMockitoMockStaticToMockito.java

```diff
@@ -327,154 +297,240 @@ private static List<Expression> getMockedTypesFromPrepareForTestAnnotation(List<
+                                                                String methodAnnotationToAdd,
+                                                                String additionalClasspathResource, String importToAdd,
+                                                                @NonNull String methodAnnotationParameters) {
+            if (hasMethodWithAnnotation(classDecl, new AnnotationMatcher(methodAnnotationSignature))) {
+                return classDecl;
             }
-            return super.visitMethodInvocation(method, ctx);
+
+            J.MethodDeclaration firstTestMethod = getFirstTestMethod(
+              classDecl.getBody().getStatements().stream().filter(statement -> statement instanceof J.MethodDeclaration)
+                .map(J.MethodDeclaration.class::cast).collect(Collectors.toList()));
+
+            JavaCoordinates tearDownCoordinates = (firstTestMethod != null) ?
+              firstTestMethod.getCoordinates().before() :
+              classDecl.getBody().getCoordinates().lastStatement();
+            classDecl = classDecl.withBody(classDecl.getBody()
+              .withTemplate(
+                JavaTemplate.builder(() -> getCursor().getParentTreeCursor(),
+                    methodAnnotationToAdd + methodAnnotationParameters + " void " + methodName + "() {}")
+                  .javaParser(() -> JavaParser.fromJavaVersion()
+                    .classpathFromResources(ctx, additionalClasspathResource)
+                    .build())
+                  .imports(importToAdd)
+                  .build(),
+                tearDownCoordinates));
+            maybeAddImport(importToAdd);
+            return classDecl;
+        }
+
+        @NotNull
+        private J.ClassDeclaration maybeAddTearDownMethodBody(J.ClassDeclaration classDecl, ExecutionContext ctx) {
+            return maybeAddMethodWithAnnotation(classDecl, ctx, "tearDownStaticMocks",
+              tearDownMethodAnnotationSignature,
+              tearDownMethodAnnotation,
+              additionalClasspathResource, tearDownImportToAdd, tearDownMethodAnnotationParameters);
         }
 
         @NotNull
         private J.MethodInvocation modifyWhenMethodInvocation(J.MethodInvocation whenMethod) {
             List<Expression> methodArguments = whenMethod.getArguments();
             List<J.MethodInvocation> staticMethodInvocationsInArguments = methodArguments.stream()
-                    .filter(expression -> expression instanceof J.MethodInvocation).map(J.MethodInvocation.class::cast)
-                    .filter(methodInvocation -> !MOCKITO_STATIC_METHOD_MATCHER.matches(methodInvocation))
-                    .filter(methodInvocation -> methodInvocation.getMethodType() != null)
-                    .filter(methodInvocation -> methodInvocation.getMethodType().hasFlags(Flag.Static))
-                    .collect(Collectors.toList());
+              .filter(expression -> expression instanceof J.MethodInvocation).map(J.MethodInvocation.class::cast)
```

Comment:
Very minor: Can be replaced with method reference `J.MethodInvocation.class::instanceOf`